### PR TITLE
container: fix quoted parameter passing to container build

### PR DESCRIPTION
**Pull request by rgerhards**
PR now contains a bit more, see individual commits. Still think all of these features make sense together, and sparing time to do separate PRs (that would need to build on each other).

### CODE REVIEW

**Comment by rgerhards** [comment]
see https://github.com/rsyslog/rsyslog-website/pull/36 for the use case

File: tools/buildenv/tools/build-doc [hunk 1]
```diff
@@ -14,7 +14,10 @@ else
 	export OUTPUT=build
 fi
 cd /rsyslog-doc
-sphinx-build $STRICT -b $FORMAT $SPHINX_EXTRA_OPTS source $OUTPUT
+if [ -e SPHINX_EXTRA_OPTS ]; then
+	export SPHINX_EXTRA_OPTS="$SPHINX_EXTRA_OPTS `cat $SPHINX_EXTRA_OPTS`"
```
Comment (deoren) [inline tools/buildenv/tools/build-doc]: Did you mean this instead?

```diff
-export SPHINX_EXTRA_OPTS="$SPHINX_EXTRA_OPTS `cat $SPHINX_EXTRA_OPTS`"
+export SPHINX_EXTRA_OPTS="$SPHINX_EXTRA_OPTS `cat SPHINX_EXTRA_OPTS`"
```

Comment (rgerhards) [reply tools/buildenv/tools/build-doc]: ohhh... I think so. Just seen that the web site also does not render correctly - might be related ;-)

Comment (deoren) [reply tools/buildenv/tools/build-doc]: @rgerhards Could be. :)

Comment (rgerhards) [reply tools/buildenv/tools/build-doc]: oh, I actually had this on my local copy ... guess I get to the point where juggling with too many things became counter-productive ;-)

**Pull request merged**